library/lms.c

```diff
@@ -499,13 +520,20 @@ static int get_merkle_path( mbedtls_lms_private_t *ctx,
                             unsigned int leaf_node_id,
                             unsigned char *path )
 {
-    unsigned char tree[MERKLE_TREE_NODE_AM_MAX][MBEDTLS_LMS_M_NODE_BYTES_MAX];
     unsigned int curr_node_id = leaf_node_id;
     unsigned int adjacent_node_id;
     unsigned int height;
     int ret = MBEDTLS_ERR_ERROR_CORRUPTION_DETECTED;
+    unsigned char *tree = NULL;
 
-    ret = calculate_merkle_tree( ctx, ( unsigned char * )tree );
+    tree = mbedtls_calloc( MERKLE_TREE_NODE_AM(ctx->params.type),
```

Comment:
MSVC says (on fe998d1d7a186fd303c06412c5a579dcc4310f15):

```
library\lms.c(529): warning C4334: '<<' : result of 32-bit shift implicitly converted to 64 bits (was 64-bit shift intended?)
```

Also at line 719. The code is correct, but we need to silence the warning. Since `MERKLE_TREE_NODE_AM` is only ever used in a context that expects `size_t`, I guess doing `(size_t)1 << …` instead of `1 << …` would work.

Reply:
Done by rebasing onto other change